MAISTRA-1079: Replace deprecated Kubernetes API's

jwendell opened the pull request.

jwendell committed:
MAISTRA-1079: Replace deprecated Kubernetes API's

dgn commented:
You forgot to include the generated charts. Running `make update-1.0-charts` will do

jwendell committed:
Updated charts

With the command `HELM_DIR="$(pwd)/resources/helm/v1.0" ./tmp/build/download-charts.sh`

jwendell commented:
> You forgot to include the generated charts. Running `make update-1.0-charts` will do

Done.

dgn commented on resources/helm/v1.0/istio/charts/mixer/templates/deployment.yaml:
```diff
@@ -58,13 +58,8 @@
           - --trace_zipkin_url=http://zipkin:9411/api/v1/spans
           {{- end }}
-        env:
-        - name: POD_NAMESPACE
```
strange. why is this being deleted? mixer won't start without this

dgn commented on resources/helm/v1.0/istio/charts/security/templates/tests/test-citadel-connection.yaml:
```diff
@@ -22,7 +22,7 @@ spec:
     - name: "{{ template "security.fullname" . }}-test"
       image: {{ .Values.global.hub }}/{{ .Values.global.proxy.image }}:{{ .Values.global.tag }}
       imagePullPolicy: "{{ .Values.global.imagePullPolicy }}"
-      command: ['sh', '-c', 'for i in 1 2 3; do curl http://istio-citadel:{{ .Values.global.monitoringPort }}/version && exit 0 || sleep 15; done; exit 1']
```
this doesn't seem to be part of this change either?

dgn commented on resources/helm/v1.0/istio/charts/security/templates/deployment.yaml:
```diff
@@ -44,9 +44,6 @@ spec:
           image: "{{ .Values.global.hub }}/{{ .Values.image }}:{{ .Values.global.tag }}"
           imagePullPolicy: {{ .Values.global.imagePullPolicy }}
           args:
-          {{- if .Values.global.sds.enabled }}
```
?

dgn reviewed (commented):
Some weird things are showing up in the changeset now. Did we possibly forget to run the patch scripts for the last release?

jwendell commented:
> Some weird things are showing up in the changeset now. Did we possibly forget to run the patch scripts for the last release?

Yeah, saw those too, left them in the PR on purpose, I thought there were changes in maistra/istio.

I'll remove them by hand now for this PR's sake and we should investigate what happened.

jwendell committed:
Clean garbage

jwendell commented:
@dgn Found the issue! I had a previously downloaded tarball (I have no idea of its age) and thus download-charts.sh was not downloading it again. Removed this old file, ran the script again and it worked fine. Pushed the changes.

knrc reviewed (changes requested):
This PR seems to go beyond the immediate issues, I believe it's only the following which will likely impact us.
- apps/v1beta1
- apps/v1beta2
The other changes should be removed from this PR.

jwendell commented:
> This PR seems to go beyond the immediate issues, I believe it's only the following which will likely impact us.
> 
>     * apps/v1beta1
> 
>     * apps/v1beta2
>       The other changes should be removed from this PR.

@knrc `extensions/v1beta1` for Deployments is mandatory as well.

jwendell commented:
Will close this one for now and open a new one when https://github.com/Maistra/istio/pull/79 is merged.

rcernich commented:
To me, this PR is preferable to what's in Maistra/istio#79, as it may make future merges with upstream more difficult and we'll need to make sure we make these same changes again should the charts change.  Also, the 1.0 changes will need to be in the maistra-1.0 branch of istio-operator as well, so they get pulled into resources/helm/v1.0 properly.
